gufe/protocols/protocoldag.py

```diff
@@ -354,7 +355,7 @@ def _from_dict(cls, dct: dict):
 
 
 def execute_DAG(protocoldag: ProtocolDAG, *,
-                shared: Optional[PathLike] = None,
+                shared: Optional[ExternalStorage] = None,
```

Comment:
could we make this `Optional[Union[PathLike, ExternalStorage]]` so it "just worked" for people that don't want to have to figure out the `ExternalStorage` concept.  I'm just seeing this as another step on the quickstart learning curve.  Internally we'd just immediately convert to `FileStorage` and all is well right?